packages/edit-widgets/src/components/save-button/index.js

```diff
@@ -1,19 +1,45 @@
+/**
+ * External dependencies
+ */
+import { filter, map, some, forEach } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import { Button } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { useDispatch } from '@wordpress/data';
-import { useState, useCallback } from '@wordpress/element';
+import { useCallback } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
 
 function SaveButton() {
-	const [ isSaving, setIsSaving ] = useState( false );
-	const { saveWidgetAreas } = useDispatch( 'core/edit-widgets' );
-	const onClick = useCallback( async () => {
-		setIsSaving( true );
-		await saveWidgetAreas();
-		setIsSaving( false );
-	}, [] );
+	const { widgetAreaEdits, isSaving } = useSelect(
+		( select ) => {
+			const {
+				hasEditsForEntityRecord,
+				isSavingEntityRecord,
+				getEntityRecords,
+			} = select( 'core' );
+			const widgetAreas = getEntityRecords( 'root', 'widgetArea' );
+			const widgetAreaIds = map( widgetAreas, ( { id } ) => id );
+			return {
+				widgetAreaEdits: filter(
+					widgetAreaIds,
+					( id ) => hasEditsForEntityRecord( 'root', 'widgetArea', id )
+				),
+				isSaving: some(
+					widgetAreaIds,
+					( id ) => isSavingEntityRecord( 'root', 'widgetArea', id )
+				),
+			};
+		}
+	);
```

Comment:
We need an empty dependency array here. Otherwise `widgetAreaEdits` will always be a new object, and in turn, `onClick` will always be a new function.